### ISSUE TITLE
Error: must pass a package name, “DLLInfo” or “DllInfoReference” object

### ISSUE THREAD

**Issue by pat-s** [r-lib/roxygen2#822]
The recent v6.1.1 update causes the following error on Travis:

https://travis-ci.org/mlr-org/mlr/jobs/451819776#L2685

```r
Running script: step_run_code(devtools::document())
Updating mlr documentation
Updating roxygen version in /home/travis/build/mlr-org/mlr/DESCRIPTION
Writing NAMESPACE
Loading mlr
Loading required package: ParamHelpers
Error: must pass a package name, “DLLInfo” or “DllInfoReference” object
```

The error does not appear with v6.1.0. Also it just appeared today after v6.1.1 was released on CRAN.

Using `devtools::document(".")` works in some instances. I think this could be related to the change made in #704?

**Comment by mb706** [comment]
The problem is [this commit](https://github.com/klutometis/roxygen/commit/9adcc8ecb874a802e6947bcd0bd65ff771704a43): dynamic libraries are not compiled, even if they don't exist. It should probably be `compile = NA` instead of `compile = FALSE`.

**Comment by pat-s** [comment]
In the mean time I'll just do a `remotes::install_version("roxygen2", "6.1.0")` before the problematic call.

**Comment by mb706** [comment]
Maybe you could make `load_all` skip `load_dll` and hope that loading the package never ends up calling a compiled function (should be rare enough, this is the old behaviour?). Otherwise (as I understand it) `roxygenise()` will just not work when no compiled dlls are present, and #771 is back where it started: compilation is necessary before documentation, only now it needs to be invoked explicitly.

**Comment by hadley** [comment]
My expectation is that most people are documenting after a `load_all()` so this should rarely be a problem. The root cause of this issue was fixed upstream in pkgdown: deployment your package website shouldn't trigger a re-roxygenise.

I agree that it would be nice if pkgload failed more gracefully if `compile = FALSE` and no dll is present.

**Comment by mb706** [comment]
I think #771 & #776, or people like them, would be happiest if documentation could be created without the necessity to build dlls at all, whether done by `roxygenise()` itself or required to do manually. I guess manual means one can skip the build process when `git checkout` changes source files and `compile = NA` would recompile unnecessarily in this case.

**Comment by eddelbuettel** [comment]
This is clearly a continuation of #771 which I filed months ago, and which was closed 17 days ago.

It is however _not fixed_.   So I have to continue to use version 6.0.1 which to me is the last non-borked one for the sole use case I have here.  Which is creating Rd files from R files.  Which is what this package was about at some point.

**Comment by hadley** [comment]
This is a continuation of #784, where I asked for help testing the feature, and no one responded. Obviously, roxygen2 works just fine for me, so I need help from the community.

@krlmlr that would seem to me to be a weird abuse of pkgload. It shouldn't create simulated packages that can't work. I think the approach that you outline in 
https://github.com/klutometis/roxygen/issues/771#issuecomment-437622036 is a better one.

**Comment by krlmlr** [comment]
We can end up in a broken state too with `compile = FALSE`. Not loading the DLL feels like a more consistent approach.

What do we need the loaded package for? `\usage{}`, `NAMESPACE`, ...? Anything where it's important to have the DLL loaded?

**Comment by hadley** [comment]
@markvanderloo I have been making these changes in response to user requests. It's not my intent to break your code. However, your comment would be more helpful if you could describe exactly what your workflow is, so I could avoid disrupting it in the future. I can not help you if you do not tell me what the problem is.

**Comment by eddelbuettel** [comment]
Any use of `roxygenize(".", roclets="rd")` on a package with `src/` does.  

And I said so over three months ago in #771.    And it is still broken, and I am still forced to rely on 6.0.1.

**Comment by hadley** [comment]
@eddelbuettel Does what?

**Comment by hadley** [comment]
Just to be absolutely sure I'm not confused, I just re-installed roxygen2, and ran `roxgenise()` directly on the vctrs package (which does have a src directory):

```R
> roxygen2::roxygenise()
Updating roxygen version in /Users/hadley/Documents/r-lib/vctrs/DESCRIPTION
Writing NAMESPACE
Loading vctrs
Writing NAMESPACE
Writing vctrs-package.Rd
```

It also works if I do `roxygen2::roxygenise(roclets = "rd")`, although I get a lot of warnings about unknown `@export` directives.

**Comment by krlmlr** [comment]
I'm seeing errors when running `roxygenise()` on a fresh clone of `vctrs`, before running `devtools::load_all()`. After running `load_all()` once I can `roxygenise()` in a fresh session.

**Comment by eddelbuettel** [comment]
Why do you waste your (and my) time dragging me into a debate about this?  

I reported it in #771 on August 3 (!!), and showed how I triggered it (calling a simple and standard `roxygenize(".", roclets="rd").`).  I have a workaround (see below) with a hard-wiring roxygen2 6.0.1.  But if I do not use that I get:

```sh
rob:~/git/rquantlib(master)$ Rscript -e 'library(roxygen2); roxygenize(".", roclets="rd")'
Updating roxygen version in /home/edd/git/rquantlib/DESCRIPTION
Loading RQuantLib
Error in getDLLRegisteredRoutines.DLLInfo(dll, addNames = FALSE) :
  must specify DLL via a “DLLInfo” object. See getLoadedDLLs()
Calls: roxygenize ... assignNativeRoutines -> getDLLRegisteredRoutines.DLLInfo
In addition: Warning message:
roxygen2 requires Encoding: UTF-8
Execution halted
rob:~/git/rquantlib(master)$
```

For comparison, here is a call that works following by the same one that fails:

```sh
rob:~/git/rquantlib(master)$ roxy.r
** Using cached version 6.0.1 of roxygen2.
Loading required package: methods
Loading required package: Rcpp
Loading required package: stats
Loading required package: graphics
Loading required package: zoo

Attaching package: ‘zoo’

The following objects are masked from ‘package:base’:

    as.Date, as.Date.numeric

rob:~/git/rquantlib(master)$ Rscript -e 'library(roxygen2); roxygenize(".", roclets="rd")'
Updating roxygen version in /home/edd/git/rquantlib/DESCRIPTION
Loading RQuantLib
Error in getDLLRegisteredRoutines.DLLInfo(dll, addNames = FALSE) :
  must specify DLL via a “DLLInfo” object. See getLoadedDLLs()
Calls: roxygenize ... assignNativeRoutines -> getDLLRegisteredRoutines.DLLInfo
In addition: Warning message:
roxygen2 requires Encoding: UTF-8
Execution halted
rob:~/git/rquantlib(master)$
```

**Comment by markvanderloo** [comment]
@hadley I generate documentation build time. Usually using a custom document.r executable script, sometimes with some options. e.g I like to generate the pdf manual using R CMD Rd2pdf while reviewing and updating it. I do a lot of coding in vim + bash with a few helper scripts or makefiles. (I do interactive data analyses in RStudio, but package coding a lot less).

not sure if this helps. Doing a `load_all()` before `roxygenize()` is a workaround.

**Comment by eddelbuettel** [comment]
I should add that it does not hit every package every time, but it is frequent enough (and I sometimes have to disable roxygen use in an RStudio profile as the error above gets triggered).  I don't know exactly what triggers it, and the roxygen2 code is on the complicated side of things so I didn't find a hook.

**Comment by hadley** [comment]
@eddelbuettel I'm trying to get information from you that you have not previously supplied (the issue that you link to complains that roxygenise doesn't work and links to a script but does not provide any output). If you think this is a waste of your time and you're happy with your work around, there's no need to continue this discussion.

@markvanderloo I said `load_all()`, but my assumption was broader: that you had a compiled shared object in your `src/`directory. I was thinking that  if you didn't use `load_all()` that you'd instead be using `R CMD install .` which also leaves compiled artefacts in `src/`. But that's obviously not happening with your workflow, which leads me to wonder what you mean by "build" time — obviously the package itself is not yet built, otherwise (I'd think) that the DLL/shared object would be available.

**Comment by eddelbuettel** [comment]
Here it is, again, on a smaller repo.  It does not always happen, but when it happens it is deadly.

```sh
edd@brad:~/git/anytime(master)$ Rscript -e 'library(roxygen2); roxygenize(".", roclets="rd")'
Updating roxygen version in /home/edd/git/anytime/DESCRIPTION
Loading anytime
Error in getDLLRegisteredRoutines.DLLInfo(dll, addNames = FALSE) : 
  must specify DLL via a “DLLInfo” object. See getLoadedDLLs()
Calls: roxygenize ... assignNativeRoutines -> getDLLRegisteredRoutines.DLLInfo
Execution halted
edd@brad:~/git/anytime(master)$ 
```

**Comment by jpritikin** [comment]
OpenMx is exhibiting a similar failure on Travis, https://travis-ci.org/OpenMx/OpenMx/jobs/455578111

Is there a workaround? I didn't read all the comments in this thread yet.

**Comment by eddelbuettel** [comment]
Yes, you could do what [roxy.r](https://github.com/eddelbuettel/littler/blob/master/inst/examples/roxy.r) does and rely on a non-borked version of `roxygen2`.  

For Travis you could (must ?) slurp in the archived 6.0.1 version.

**Comment by hadley** [comment]
@jpritikin you can either call `pkgload::load_all()` or `R CMD install` in order to compile the DLL first.

**Comment by jpritikin** [comment]
@hadley Wow, this worked great. Thanks!

**Comment by RLumSK** [comment]
Just to feedback and contribute to this discussion. Since 2015, I sucessfully used the following line  
to create the package documentation

```r
devtools::document(pkg = ".", roclets = NULL)
```

This worked until I updated to 'roxygen2' v6.1.1. Then I got

`
Error in getDLLRegisteredRoutines.DLLInfo(dll, addNames = FALSE) : 
  must specify DLL via a “DLLInfo” object. See getLoadedDLLs()
`

Now calling 

```r
pkgbuild::compile_dll()
devtools::document()
```

(so basically what was written above) provides the workaround.

**Comment by rnuske** [comment]
Following the steps detailed in [Usage Section](https://github.com/r-lib/usethis/blob/master/README.md#usage) of the current `usethis` dev  (commit 85bf30a) up to `use_rcpp()` followed by `devtools::document()` causes this error as well.

```
> devtools::document()
Updating mypkg documentation
Loading mypkg
Fehler in getDLLRegisteredRoutines.DLLInfo(dll, addNames = FALSE) : 
  must specify DLL via a “DLLInfo” object. See getLoadedDLLs()
```

<Details><summary>session_info()</summary>

```r
> devtools::session_info()
─ Session info ─────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.4.4 (2018-03-15)
 os       Ubuntu 16.04.5 LTS          
 system   x86_64, linux-gnu           
 ui       RStudio                     
 language (EN)                        
 collate  de_DE.UTF-8                 
 ctype    de_DE.UTF-8                 
 tz       Europe/Berlin               
 date     2019-01-03                  

─ Packages ─────────────────────────────────────────────────────────────────────
 ! package     * version    date       lib source                        
   assertthat    0.2.0      2017-04-11 [1] CRAN (R 3.4.4)                
   backports     1.1.3      2018-12-14 [1] CRAN (R 3.4.4)                
   callr         3.1.1      2018-12-21 [1] CRAN (R 3.4.4)                
   cli           1.0.1      2018-09-25 [1] CRAN (R 3.4.4)                
   commonmark    1.7        2018-12-01 [1] CRAN (R 3.4.4)                
   crayon        1.3.4      2017-09-16 [1] CRAN (R 3.4.4)                
   desc          1.2.0      2018-05-01 [1] CRAN (R 3.4.4)                
   devtools    * 2.0.1      2018-10-26 [1] CRAN (R 3.4.4)                
   digest        0.6.18     2018-10-10 [1] CRAN (R 3.4.4)                
   fs            1.2.6      2018-08-23 [1] CRAN (R 3.4.4)                
   glue          1.3.0      2018-07-17 [1] CRAN (R 3.4.4)                
   magrittr      1.5        2014-11-22 [1] CRAN (R 3.4.4)                
   memoise       1.1.0      2017-04-21 [1] CRAN (R 3.4.4)                
 R mypkg         0.0.0.9000 <NA>       [?] <NA>                          
   pkgbuild      1.0.2      2018-10-16 [1] CRAN (R 3.4.4)                
   pkgload       1.0.2      2018-10-29 [1] CRAN (R 3.4.4)                
   prettyunits   1.0.2      2015-07-13 [1] CRAN (R 3.4.4)                
   processx      3.2.1      2018-12-05 [1] CRAN (R 3.4.4)                
   ps            1.3.0      2018-12-21 [1] CRAN (R 3.4.4)                
   purrr         0.2.5      2018-05-29 [1] CRAN (R 3.4.4)                
   R6            2.3.0      2018-10-04 [1] CRAN (R 3.4.4)                
   Rcpp          1.0.0      2018-11-07 [1] CRAN (R 3.4.4)                
   remotes       2.0.2      2018-10-30 [1] CRAN (R 3.4.4)                
   rlang         0.3.0.1    2018-10-25 [1] CRAN (R 3.4.4)                
   roxygen2      6.1.1      2018-11-07 [1] CRAN (R 3.4.4)                
   rprojroot     1.3-2      2018-01-03 [1] CRAN (R 3.4.4)                
   rstudioapi    0.8        2018-10-02 [1] CRAN (R 3.4.4)                
   sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 3.4.4)                
   stringi       1.2.4      2018-07-20 [1] CRAN (R 3.4.4)                
   stringr       1.3.1      2018-05-10 [1] CRAN (R 3.4.4)                
   testthat      2.0.1      2018-10-13 [1] CRAN (R 3.4.4)                
   usethis     * 1.4.0.9000 2019-01-03 [1] Github (r-lib/usethis@85bf30a)
   withr         2.1.2      2018-03-15 [1] CRAN (R 3.4.4)                
   xml2          1.2.0      2018-01-24 [1] CRAN (R 3.4.4)                

[1] /home/rnuske/R/x86_64-pc-linux-gnu-library/3.4
[2] /usr/lib/R/site-library
[3] /usr/lib/R/library

 R ── Package was removed from disk.
```

</details></br>

**Comment by jameslamb** [comment]
While working on `LightGBM` tonight, I was able to get around this issue by downgrading `devtools` to 1.13.6.

I downgraded like this:

```
Rscript -e "remove.packages('devtools')"
wget https://cran.r-project.org/src/contrib/Archive/devtools/devtools_1.13.6.tar.gz -O devtools.tar.gz
R CMD INSTALL devtools.tar.gz
```

Note from the session info below that I had roxygen 6.1.1 installed. Hopefully this comment lends evidence to the fact that the problem isn't explicitly _a roxygen2 problem_ but some inconsistency between roxygen2 v6.1.1 and devtools2.0.0.

<details><summary>session info</summary>

```
R version 3.5.0 (2018-04-23)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

other attached packages:
[1] roxygen2_6.1.1  devtools_1.13.6

loaded via a namespace (and not attached):
 [1] compiler_3.5.0    R6_2.3.0.9000     magrittr_1.5     
 [4] tools_3.5.0       withr_2.1.2.9000  yaml_2.2.0       
 [7] Rcpp_0.12.19.3    memoise_1.1.0     xml2_1.2.0       
[10] stringi_1.2.4     data.table_1.11.8 stringr_1.3.1    
[13] digest_0.6.18     commonmark_1.5
```

</details>

**Comment by patr1ckm** [comment]
I hit this while installing an old internal package. I don't know the reasons, but rolling all the way back to `roxygen 6.0.1` made it work for me.

**Comment by eddelbuettel** [comment]
@patr1ckm See my comment from Nov 15 -- I have been forced to do that ever since `roxygen` broke^"changed in this unfortunate way".   So I keep version of 6.0.1 around just for that.  Not ideal.

**Comment by krlmlr** [comment]
Can you please double-check with the development version of {pkgload}?

**Comment by RLumSK** [comment]
It looks good to me. 

Although I used the workaround as written above, I had still trouble in combination with  `pkgdown::build_site()` leading to `Error in getDLLRegisteredRoutines.DLLInfo(dll, addNames = FALSE)`.  

With the development version of 'pkgload' from GitHub this problem appears to be solved and `pkgdown::build_site()` runs without further tweaks (I tried two times with the old and the new version to be sure). 

So far at least for my case, of course, I cannot talk for the other cases reported above.

**Comment by iago-pssjd** [comment]
After some modifications on the package `openxlsx`, I was going to `Build > Check Package` RStudio commands and I got a similar output:
```
Error in getDLLRegisteredRoutines.DLLInfo(dll, addNames = FALSE) : 
  must specify DLL via a “DLLInfo” object. See getLoadedDLLs()
Calls: <Anonymous> ... assignNativeRoutines -> getDLLRegisteredRoutines.DLLInfo
```
Now, I decided to go first to `Build > Test Package`  and then try `Build > Check Package` and it seems to have worked.

**Comment by hadley** [comment]
roxygen will now compile code in src when needed, which should fix most of the problems in this thread. 

If you don't want roxygen2 to compile your source code, you have two options:

* `Roxygen: list(load = "source")` or `roxygenise(load = "source")` will use the strategy from roxygen2 6.0.0 and earlier

* `Roxygen: list(load = "installed")` or `roxygenise(load = "installed")` will use the installed version of the package, assuming that you've installed it some other way

**Comment by vtraag** [comment]
Currently, I am [working on updating](https://github.com/igraph/rigraph/pull/372) an R package (`rigraph`) that provides an interface for a C library (`igraph`).  I seem to be running in a type of catch-22 situation when running `roxygen2` on that package.

Using the `load = "source"` option leads to the problem mentioned in issue #595 about the missing `roxygen_devtest` package. The cause of this error is in 

https://github.com/igraph/rigraph/blob/5e7a689afde1864c2f36b2a7825db24f2f89b22b/R/zzz-deprecate.R#L39

where `packageName()` evaluates to `roxygen_devtest`. This could perhaps be resolved in some way, but it it not immediately clear to me how.

So, in order for `roxygen2` to properly generate documentation, I therefore need to use the `load = "installed"` option. This then requires me to call `pkgbuild::compile_dll` before calling `roxygenise(load = "installed")`, because I otherwise get the error mentioned here. However, `pkgbuild::compile_dll` requires a `NAMESPACE` file, which should actually be generated by `roxygen2`. I now solve this by first creating an artificial `NAMESPACE` file, which then needs to be removed again in order to have `roxygen2` generate the proper `NAMESPACE` file. This seems to work. However, there is a rather strange dependency here: `roxygenise` depends on `compile_dll` which depends on the output of `roxygenise` again. I imagine others have faced this situation as well. I am hoping someone will have a more "proper" solution then this one?

**Comment by eddelbuettel** [comment]
I tried to look at your package to see my "solution" of relying on the older version works, but your package / repo seems non-standard so I can't.

```sh
edd@rob:/tmp/rigraph(update/0.8.0)$ roxy.r 
** Using cached version 6.0.1 of roxygen2.
First time using roxygen2. Upgrading automatically...
Error: File `/tmp/rigraph/DESCRIPTION' does not exist
edd@rob:/tmp/rigraph(update/0.8.0)$ 
```

**Comment by vtraag** [comment]
@eddelbuettel , thanks for taking a look! The package first needs to be bootstrapped by calling `make`. This generates the `DESCRIPTION` file (among others), and somewhere during this process the `NAMESPACE` is also generated.

**Comment by eddelbuettel** [comment]
Very non-standard build. 

Something somewhere which I cannot find seems to require `roxygen_devtest`. Which I don't have.  Moving on ...

**Comment by vtraag** [comment]
Yes, indeed,  hence some of the issues that arise. Thanks for taking a look nonetheless.

Regarding the problem with `roxygen_devtest`:
>Using the `load = "source"` option leads to the problem mentioned in issue #595 about the missing `roxygen_devtest` package. The cause of this error is in
>
> https://github.com/igraph/rigraph/blob/5e7a689afde1864c2f36b2a7825db24f2f89b22b/R/zzz-deprecate.R#L39
>
> where `packageName()` evaluates to `roxygen_devtest`. This could perhaps be resolved in some way, but it it not immediately clear to me how.

If I understand it correctly, running `load="source"` is equivalent to the procedure of `roxygen2` of version 6.0.x and earlier, according to the [documentation](https://roxygen2.r-lib.org/reference/load.html). It therefore makes sense that you also run into this problem when using an older version of `roxygen2`, but good to see that confirmed, thanks!

**Comment by krlmlr** [comment]
@vtraag: Can you please try the dev version of the pkgload package?

**Comment by vtraag** [comment]
> @vtraag: Can you please try the dev version of the pkgload package?

@krlmlr , yes, I would be glad to. In what way? Using the `load = "source"` option, or using the `load = "installed"` option?

**Comment by krlmlr** [comment]
Install it via `devtools::install_github("r-lib/pkgload")`, then update documentation with roxygen2. The package is used internally by roxygen2.

**Comment by vtraag** [comment]
@krlmlr, thanks, but that unfortunately that does not solve things. 

The problem is also understandable: `roxygenize(load = "installed")` tries to load an installed package, but it is not yet installed, because it is in the process of being installed. Hence, it requires first `pkgbuild::compile_dll`, which brings the outlined catch-22 situation. In the [documentation](https://roxygen2.r-lib.org/reference/load.html) it says:

> Use this strategy if you have installed a development version of the package already. This is the highest fidelity strategy, but requires work outside of roxygen2.

Does the "work outside of roxygen2" here effectively refer to the type of catch-22 situation that I outlined?

The alternative of using `roxygenize(load = "source")` also still yields the same error of missing `roxygen_devtest`.

**Comment by gaborcsardi** [comment]
Running 
```
git clean -fdx .
make
```

builds the R package, including the docs just fine.

**Comment by vtraag** [comment]
The problem is that the generated `NAMESPACE` is incomplete, so that the functions are not correctly exported. That is at least the problem that I experienced when testing locally (before trying it on Travis), because it errors out at some point with the `DLLInfo` problem reported in this issue. I'm now [double checking](https://travis-ci.org/github/igraph/rigraph/builds/661522562?utm_source=github_status&utm_medium=notification) whether this also does not work on Travis. Can you verify whether the `NAMESPACE` you generate is correct, @gaborcsardi ? If so, it begs the question how your local setup is different from mine (and Travis, presumably).

Note that the workaround of creating a temporary dummy `NAMESPACE` file does work on [Travis](https://travis-ci.org/github/igraph/rigraph/builds/661481268?utm_source=github_status&utm_medium=notification) (note that there are still some other issues that result in a failed build, which can be resolved, and on which I am working now).

**Comment by gaborcsardi** [comment]
This does not really belong into a closed issue of roxygen2.... but there is no setup or anything:

```
git clone https://github.com/igraph/rigraph.git
cd rigraph
git submodule init
git submodule update
make
R CMD INSTALL igraph_1.1.2.9000.tar.gz
```

**Comment by vtraag** [comment]
You are right: `make` does seem to run correctly on [Travis](https://travis-ci.org/github/igraph/rigraph/jobs/661522563?utm_medium=notification&utm_source=github_status), at least for R 3.5. It does not on my machine (also R 3.5), however, and I therefore assumed it would neither on Travis. The output on my local machine (using `git clean -fdx` followed by `make`) produces:

```
First time using roxygen2. Upgrading automatically...
Updating roxygen version in /tmp/tmp.RWJYtN0M6u/DESCRIPTION
Writing NAMESPACE
Loading igraph
Error in getDLLRegisteredRoutines.DLLInfo(dll, addNames = FALSE) : 
  must specify DLL via a “DLLInfo” object. See getLoadedDLLs()
Calls: document ... assignNativeRoutines -> getDLLRegisteredRoutines.DLLInfo
Execution halted
Rscript -e 'devtools::build(path = ".")'
✔  checking for file ‘/home/vtraag/Dropbox/traveller/src/rigraph/DESCRIPTION’ ...
─  preparing ‘igraph’:
✔  checking DESCRIPTION meta-information ...
─  cleaning src
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories (382ms)
─  building ‘igraph_1.1.2.9000.tar.gz’
```

Notice the `DLLInfo` issue reported here, hence I considered this to be the appropriate place to discuss this. Anyway, I will investigate further what the difference is between my local machine and the setup on Travis.

**Comment by gaborcsardi** [comment]
Try to run the make steps one by one, and then go into the temporary directory (`/tmp/tmp.RWJYtN0M6u` above) and see what is wrong there. 

I suspect that your tree is not functional, is this the `dev` branch as on GitHub or have you modified it?

**Comment by vtraag** [comment]
The problem appears with `roxygen2` version `7.0.2`. The released version of `7.1.0`, which was released **yesterday**, solves the issue. I was testing this already *before* yesterday, hence, the problem seemed to be related to this issue. Anyway, glad to see it is resolved in version `7.1.0`.

**Comment by vtraag** [comment]
@gaborcsardi , I can try to see if I can debug the issue further for `roxygen2`, version `7.0.2`, but given it is resolved in version `7.1.0`, perhaps that is no longer relevant. Let me know if you want me to take another look at the issue.

**Comment by gaborcsardi** [comment]
Oh no, if it works with the new version, that's fine.
